torch/_inductor/bounds.py

```diff
@@ -19,7 +19,7 @@ class BoundVars:
     def __init__(self, loop_body: LoopBody):
         self.loop_body = loop_body
         self.replacement_vals = {
-            k: ValueRanges(0, v if not free_symbols(v) else math.inf)
+            k: ValueRanges(0, v - 1 if not free_symbols(v) else math.inf)
```

Comment:
@ezyang should this be
```suggestion
            k: ValueRanges(0, v - 1) if not free_symbols(v) else ValueRanges(2, math.inf)
```
?

Reply:
The >= 2 assumption was used to reduce amount of (empirically) unnecessary guards. I don't think the same logic applies here, might affect correctness of generating code.

Reply:
Can we have a dynamic size of size 1? I thought it would be specialised.

Reply:
I don't really know what `replacement_vals` here is used for, so I am deferring to @eellison

Reply:
I think you can compile on dynamic, then pass in 1 later. That use case is why we have the assumption of >2 in tracing, to avoid recompilation, afaik.

Reply:
If you pass a 0 or a 1 you recompile. Printing `loop_body.var_ranges` here and executing
```python
@torch.compile(dynamic=True)
def fn(a, b):
    return a[b]

x = torch.randn(8)
b = torch.tensor([1,1,2])
fn(x, b)
b = torch.zeros(2).long()
fn(x, b)
b = torch.zeros(1).long()
fn(x, b)
```
prints:
```
{z0: s1}
{}
```
i.e. first compilation, no recompilation with tensor of size 2, then compile again with tensor of size 1 (specified).

Reply:
You're right. I printed out the guards and I see the >=2 guards. 

I think this is true to the extent that we are not doing aot_inductor export. @desertfire do you know what the plan is for exporting wrt/guards on inputs, do we just ignore them ? do we have any plans to do multiple compilations for different guards ?